tests/src/tools/ReadyToRun.SuperIlc/Commands/CompileFromCrossgenRspCommand.cs

```diff
@@ -0,0 +1,271 @@
+                return 1;
+            }
+
+            if (options.RewriteOldPath != null && options.RewriteNewPath != null)
+            {
+                if (options.RewriteOldPath.Length != options.RewriteNewPath.Length)
+                {
+                    Console.Error.WriteLine("Error: --rewrite-old-path and --rewrite-new-path were specified a different number of times.");
+                    return 1;
+                }
+
+                for (int i = 0; i < options.RewriteNewPath.Length; i++)
+                {
+                    pathReplacements.Add(options.RewriteOldPath[i].FullName, options.RewriteNewPath[i].FullName);
+                    Console.WriteLine($"Re-writing path {options.RewriteOldPath[i].FullName} as {options.RewriteNewPath[i].FullName}");
+                }
+            }
+
+            bool success = true;
+            int compilationFailures = 0;
+            int totalCompilations = 0;
+            // Collect all the compilations first
+            foreach (var inputRsp in responseFiles)
+            {
+                var crossgenArguments = CrossgenArguments.ParseFromResponseFile(inputRsp)
+                                                         .ReplacePaths(pathReplacements);
+
+                Console.WriteLine($"{inputRsp} -> {crossgenArguments.InputFile}");
+                var compilerRunners = GetCompilerRunners(options, crossgenArguments.ReferencePaths);
+
+                string responseFileOuputPath = Path.Combine(options.OutputDirectory.FullName, Path.GetFileNameWithoutExtension(inputRsp));
+                responseFileOuputPath.RecreateDirectory();
+
+                List<ProcessInfo> fileCompilations = new List<ProcessInfo>();
+                foreach (CompilerRunner runner in compilerRunners)
+                {
+                    var compilationProcess = new ProcessInfo(new CompilationProcessConstructor(runner, responseFileOuputPath, crossgenArguments.InputFile));
+                    fileCompilations.Add(compilationProcess);
+                }
+
+                ParallelRunner.Run(fileCompilations, options.DegreeOfParallelism);
+                totalCompilations++;
+
+                foreach (var compilationProcess in fileCompilations)
+                {
+                    if (!compilationProcess.Succeeded)
+                    {
+                        success = false;
+                        compilationFailures++;
+
+                        Console.WriteLine($"Failed compiling {compilationProcess.Parameters.InputFileName}");
+                    }
+                }
+            }
+
+            Console.WriteLine("Rsp Compilation Results");
+            Console.WriteLine($"Total compilations: {totalCompilations}");
+            Console.WriteLine($"Compilation failures: {compilationFailures}");
+
+            return success ? 0 : 1;
+        }
+
+        private static IEnumerable<CompilerRunner> GetCompilerRunners(BuildOptions options, IEnumerable<string> referencePaths)
```

Comment:
Can you please clarify the relationship between this method and the existing BuildOptions.CompilerRunners? Ideally I would love us to be able to unify them somehow, perhaps by adding a new parameter to the BuildOptions method if necessary.

Reply:
I needed custom reference paths because each response file may have different ones so we cannot rely on the single set in BuildOptions. Now you mention it, we can just add an override to `BuildOptions.CompilerRunners` and unify :)